### DISCUSSION TITLE
crossing waters tool gives absurd number of crossings

### DISCUSSION THREAD

**Discussion by StochasticCactus** [GrossfieldLab/loos#119] [Q&A] [answered]
Hi, 

I'm trying to analyse the number of waters crossing a membrane containing a water pore, in a 2 µs simulation from Gromacs. I'm using the crossing-waters tool. So I grouped the output by every 5 frames, which gives me the number of waters crossing the membrane per nano second. But I got absurd results. 

First I got that the rate of waters crossing was constant in the range of 10 to 40 crossings per ns, but I got the same values for my control membrane, which is absurd.

So, I tried to process my trajectory in another way, but this time got results in the thousands crossings per ns.

![crossing_events](https://github.com/GrossfieldLab/loos/assets/60858350/97e2596e-da9a-4c80-8ea4-2c06b36bbdb2)

I thought I could be processing the output wrong, but the output has 496632 lines, so I'm pretty sure it's a problem oin centralizing the membrane or something related to the trajectory, as I can't even see waters crossing the membrane when visualizing the trajectory. 

I tried porcessing my trajectory by the following ways:

`gmx trjconv -f md.xtc -s md.tpr -pbc mol -center -o md_centered.xtc`

`gmx trjconv -f control.xtc -s control.tpr -pbc whole -o control_whole.xtc -center`

and generated my gro file for the crossing waters tool  using the `-dump 0` option form `gmx trjconv`.

additionally I tried centralizing the gro file using `gmx editconf`

I'm running the crossing-waters with the following comand:

`crossing-waters control_f0.gro control_whole.xtc 10 20 'resname == SOL && !hydrogen' > crossing_waters_control.out `

I really need to finish these results until next monday. 

Any help will be appreciated.

Thanks

**Comment by agrossfield** [comment, marked as answer]
When you look at your system in vmd or whatever, is the membrane centered? If not (and especially if the bilayer is centered at the periodic boundary) you'll get ridiculous results, because loos assumes you've pre-curated your data.

I don't know much about the gromacs toolset, but both of those commands look like they'll center the whole system, which doesn't do you much good, since that doesn't place the membrane at z=0.

I suggest you take a look at 2 wiki entries to help you. One deals with membrane simulations (https://github.com/GrossfieldLab/loos/wiki/Analyzing-membrane-simulations) and the other with gromacs (https://github.com/GrossfieldLab/loos/wiki/Preparing-to-analyze-a-gromacs-simulation). Taken together, the two would recommend you first generate a fake psf (gro files don't have connectivity, which `merge-traj` relies on), then curate the data with something like:

```
merge-traj --fix-imaging 1 --skip-first-frame 1 --z-centering-selection 'resname == "SOMETHING THAT SELECTS LIPID"' --selection-is-split 1 FAKE.psf fixed.dcd control_whole.xtc
```

If you're working with gromacs's original xtc file, you might also need the `--fix-imaging` flag, which fixes molecules that gromacs wrote broken across the periodic box.

Once that's done, rerun `crossing-waters` with the new psf and dcd, and it should work. 

Hope this helps,

Alan

**Reply by StochasticCactus** [reply]
Hi, 

Thank you very much for your reply.

I still don't understand why `gmx trjconv` didn't work as it allows you to center a selection. But after running it with `merge-traj` it finally worked.

Thanks again,

Tarcillo
